pkg/cmd/cli/cmd/rollout/rollout.go

```diff
@@ -134,3 +137,40 @@ func NewCmdRolloutUndo(fullName string, f *clientcmd.Factory, out io.Writer) *co
 	cmd.Example = fmt.Sprintf(rolloutUndoExample, fullName)
 	return cmd
 }
+
+const (
+	rolloutCancelLong = `
+Cancel the in-progress deployment
+
+Running this command will cause the current in-progress deployment to be
+cancelled, but keep in mind that this is a best-effort operation and may take
+some time to complete. It’s possible the deployment will partially or totally
+complete before the cancellation is effective. In such a case an appropriate
+event will be emitted.
+`
+
+	rolloutCancelExample = `  # Cancel the in-progress deployment based on 'nginx'
+  %[1]s rollout cancel dc/nginx`
+)
+
+func NewCmdRolloutCancel(fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	opts := &RolloutCancelOptions{}
+	return &cobra.Command{
+		Use:     "cancel (TYPE NAME | TYPE/NAME) [flags]",
+		Long:    rolloutCancelLong,
+		Example: fmt.Sprintf(rolloutCancelExample, fullName),
+		Short:   "cancel the in-progress deployment",
+		Run: func(cmd *cobra.Command, args []string) {
+			allErrs := []error{}
+			err := opts.Complete(f, args, out)
+			if err != nil {
+				allErrs = append(allErrs, err)
```

Comment:
We shouldn't do this. That's the problem you need to fix for `rollout pause` and `rollout resume` upstream in https://github.com/kubernetes/kubernetes/pull/31234